### PR TITLE
Add EAGLE-style MTP module support for DeepSeek-R1 and example usage script

### PR DESCRIPTION
**Pull request by benchislett**
This is @CentML's implementation of DeepSeek MTP modules that enable speculative decoding for DeepSeek-R1.

The changes in this branch enable `--speculative-model DeepSeekV3MTP` to register itself as an alternate implementation of DeepSeek-R1, which load only the MTP weights and have modified model code to invoke only the MTP layer. This model code resides in `deepseek_mtp.py`.

An additional change moves the RMSNorm application of the final hidden states into the `compute_logits` function, such that the `previous_hidden_states` which are taken from the output of `model(...)` are un-normalized. Experimental results show a small increase in predictive accuracy by making this change, which makes sense intuitively as the hidden states are treated as they would be for an additional layer of the transformer model and the output norm is treated separately as a part of the output head.

This code also enables the EAGLE code path for reusing `previous_hidden_states` across TP workers in the base model runner code, enabling (single-step) multi-GPU draft model execution.

While it is possible that the hidden states can be reused from each worker and do not need to be broadcasted to the TP workers between iterations, we choose to follow the EAGLE syntax and respect the abstraction boundary of draft worker / target worker for speculative decoding.

Notably, this implementation does not mask the zero-th position of the input embeddings into the MTP module, though the existing EAGLE models do. This is because of an unknown issue with CUDA graphs and MLA attention that cause accuracy issues when this is performed. Anecdotal testing shows much improved acceptance rates by omitting this problematic masking. 

Note that this is not the first vLLM PR to implement MTP module support for DeepSeekV3 models, and serves primarily as a reference implementation for validation purposes. See [vllm#12755](https://github.com/vllm-project/vllm/pull/12755)

### CODE REVIEW

**Comment by benchislett** [comment]
PR made and reviewed in upstream, closing this draft.
